### DISCUSSION TITLE
Failed to acquire required resources.

### DISCUSSION THREAD

**Discussion by thatjade** [Alexays/Waybar#1826] [Q&A]
I cant start Waybar on Gnome with Mutter, hence all the config/styles are in place.

![image](https://user-images.githubusercontent.com/77240684/114315872-48961500-9b01-11eb-9ff9-832a087b1e3c.png)
![image](https://user-images.githubusercontent.com/77240684/114315893-58155e00-9b01-11eb-8216-ce9b746fcfc0.png)

OS: ArchLinux
DE: Gnome 40
WM: Mutter

**Comment by Alexays** [comment]
What's your waybar version?

**Comment by lildata** [comment]
I do have the same error message when launching on Gnome with a very minimal waybar config file 
`[error] Failed to acquire required resources.` https://github.com/Alexays/Waybar/blob/master/src/client.cpp#L184

Gnome 40.4
waybar 0.9.8

**Comment by godalming123** [comment]
same issue but this time with weston window manager and endevour os
```
[whatever@whatever-hplaptop14cm0xxx ~]$ waybar -v
Waybar v0.9.8
[whatever@whatever-hplaptop14cm0xxx ~]$ waybar -c=.config/waybar/config  -s=.con
fig/waybar/style.css
[2021-11-20 08:49:49.846] [info] Resources files: .config/waybar/config, .config
/waybar/style.css
[2021-11-20 08:49:49.849] [error] Failed to acquire required resources.
[whatever@whatever-hplaptop14cm0xxx ~]$ ls .config/waybar
config  config.json  style.css
[whatever@whatever-hplaptop14cm0xxx ~]$ 

```

**Comment by wh1rstclass** [comment]
Same issue,  Manjaro with Gnome

```
[whatever@whatever]$ uname -a
Linux whatever 5.10.79-1-MANJARO #1 SMP PREEMPT Fri Nov 12 20:26:09 UTC 2021 x86_64 GNU/Linux
[whatever@whatever]$ echo $XDG_SESSION_TYPE
wayland
[whatever@whatever waybar]$ waybar -v
Waybar v0.9.8-81-g331dfa8 (branch 'master')
[whatever@whatever]$ waybar 
[2021-11-23 18:23:08.040] [info] Using configuration file /home/whatever/.config/waybar/config
[2021-11-23 18:23:08.040] [info] Using CSS file /home/whatever/.config/waybar/style.css
[2021-11-23 18:23:08.044] [error] Failed to acquire required resources.
```
If I try to run under sudo , assuming that the point in the insufficient permissions, I am got this:
```
[whatever@whatever waybar]$ sudo waybar
[2021-11-23 18:24:37.584] [error] Bar need to run under Wayland
```
from root
```
[whatever ~]# waybar -c /home/whatever/.config/waybar/config -s /home/whatever/.config/waybar/style.css 

(waybar:67632): Gtk-WARNING **: 19:06:35.250: cannot open display:

```
`journalct -efx` doesn't show anything when error appear 
more logging info maybe?
`waybar -l info/debug/trace` has the same output than simple `waybar`

**Comment by UNtergas** [comment]
The true problem with waybar and gnome is this:
* (waybar:104745): WARNING **: 19:01:58.738: It appears your Wayland compositor does not support the Layer Shell protocol
[2024-12-31 19:01:58.738] [error] The Wayland compositor does not support wlr-layer-shell protocol

Since Gnome use mutter which doenst support wlr-layer-shell protocol, it is basically imposible to use waybar on gnome.
If you want to use waybar it is advise to change to a different DE or Wayland compositor 
https://wayland.app/protocols/wlr-layer-shell-unstable-v1
